File: lib/底层网络知识之重要的传输层/Socket/demos/python/tcp_server.py
import socket

def tcp_server():
  # 创建 socket 对象
  serversocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

  # 获取本地主机名
  # host = socket.gethostname()
  host = '127.0.0.1'
  port = 9999

  # 绑定端口号
  serversocket.bind((host, port))

  # 设置最大连接数，超过后排队
  serversocket.listen(5)
  print('Waiting for connection...')

  while True:
    # 建立客户端连接
    clientsocket, addr = serversocket.accept()
    print(f"连接地址：{str(addr)}")
    msg = '欢迎访问TCP服务器！' + "\r\n"
    clientsocket.send(msg.encode('utf-8'))
    # 发送消息返回给客户端
    while True:
        data = clientsocket.recv(1024)
        time.sleep(1)
        if not data or data.decode('utf-8') == 'exit':
            break
        clientsocket.send(('Hello, %s!' % data.decode('utf-8')).encode('utf-8'))
    clientsocket.close()

if __name__ == '__main__':
  tcp_server()

File: lib/底层网络知识之重要的传输层/Socket/demos/nodejs/tcp_server.js
const net = require('net');

// 创建服务器
const server = net.createServer((socket) => {
  console.log('客户端已连接');

  // 接收来自客户端的数据
  socket.on('data', (data) => {
    console.log('收到客户端数据:', data.toString());
  });

  // 客户端断开连接的事件
  socket.on('end', () => {
    console.log('客户端已断开连接');
  });

  // 发送数据到客户端
  socket.write('你好，来自服务器的问候!\n');
});

// 监听 3000 端口
server.listen(3000, () => {
  console.log('服务器正在监听端口 3000');
});